app/scripts/ui.js

```diff
@@ -24,6 +26,21 @@ import metaRPCClientFactory from './lib/metaRPCClientFactory';
 start().catch(log.error);
 
 async function start() {
+  async function displayCriticalError(container, err, metamaskState) {
+    const html = await getErrorHtml(SUPPORT_LINK, metamaskState);
+
+    container.innerHTML = html;
+
+    const button = document.getElementById('critical-error-button');
+
+    button.addEventListener('click', (_) => {
+      browser.runtime.reload();
```

Comment:
Here we can all method to initialise UI again rather than reloading the browser.

Reply:
What method can we call to initialise the UI?

Reply:
> @segun Do you know why the tests are failing?

Looking into it.

Reply:
https://github.com/MetaMask/metamask-extension/pull/14461/files/af15b73ad4ca96b85ca2a75aa6b9b9040ec40bbb#diff-5a9d964db9d7c2fe207e21470884b2710e54723e0c1e6e6f7fd264706604fd6cR58 can help

Reply:
Nice that you were able to figure this out, thanks for that. The one thing about this button is that when MetaMask reloads, the extension closes, and then nothing happens. I know you can click on the popup to open MetaMask again but I think people might wait for the popup to open again. Do we know if there is a way to automatically do that? If this is something that can be added later just to get this out the door, that's fine.

Reply:
I'm thinking to have one or two extra PRs to fix some of this issue that are not necessarily blockers for this PR.

Reply:
Ok no problem.